youtube_dlc/compat.py

```diff
@@ -2365,6 +2370,20 @@ class compat_HTMLParseError(Exception):
 except NameError:
     compat_str = str
 
+try:
+    from urllib.parse import quote as compat_urllib_parse_quote
+    from urllib.parse import quote_plus as compat_urllib_parse_quote_plus
```

Comment:
The CI test problem originates here. And it has to do with a stdlib change:

> Changed in version 3.7: Moved from RFC 2396 to RFC 3986 for quoting URL strings. “~” is now included in the set of unreserved characters.

Source: https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote

The test said:

```
FAIL: test_compat_urllib_parse_quote (test.test_compat.TestCompat)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/blackjack4494/yt-dlc/test/test_compat.py", line 60, in test_compat_urllib_parse_quote

    self.assertEqual(compat_urllib_parse_quote('/~user/abc+def'), '/%7Euser/abc%2Bdef')

AssertionError: '/~user/abc%2Bdef' != '/%7Euser/abc%2Bdef'
```

And, similarly:

```
FAIL: test_compat_urllib_parse_quote_plus (test.test_compat.TestCompat)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/blackjack4494/yt-dlc/test/test_compat.py", line 77, in test_compat_urllib_parse_quote_plus

    self.assertEqual(compat_urllib_parse_quote_plus('~/abc def'), '%7E%2Fabc+def')

AssertionError: '~%2Fabc+def' != '%7E%2Fabc+def'
```

Since the app supports Python versions < v3.7 and >= v3.7 and the new `iri_to_uri()` this PR introduces is the only user of these two functions and always treats `~` as safe anyway, their tests can just exclude `~`. If we would adapt the tests to the new behavior, testing for versions < v3.7 would fail.

Reply:
Is it safe to change the test itself instead of changing the behaviour of the function? Are you sure it won't cause any side effects?

Reply:
It's an import. See "Source" link to documentation. We can't change those functions. And changing `iri_to_uri()`, if you were referring to that, wouldn't satisfy the test.

I don't see why it should be bad. The tests themselves are [introduced by this PR](https://github.com/blackjack4494/yt-dlc/pull/184/files#diff-649263b09e0fabfa2fa466e6cdc3bc4b240193d647ce1aecbfc2e54ef416a92c).

Reply:
I meant modifying the function `compat_urllib_parse_quote`. I am going to commit your changes, but I believe it will cause the tests to fail for python versions below 3.7. 

I wont have much free time for a couple days. After that I will have a better look at it

Reply:
`from urllib.parse import quote as compat_urllib_parse_quote` means `compat_urllib_parse_quote()` is the stdlib function `urllib.parse.quote()`, which is documented behind [this link](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote) from my above comment.

Reply:
You are right, my bad